src/services/parser.service.ts

```diff
@@ -28,8 +28,8 @@ export class ParserService extends AbstractService {
     });
 
     this.subscribeToFiles();
-    this.subscribeToSettings();
-    this.parseSavedDocuments();
+    await this.subscribeToSettings();
+    await this.parseSavedDocuments();
```

Comment:
It won't wait because subscription is a procedure, not function :) So I don't think so that this is needed.

Reply:
SonarCloud complains about this being missing.